src/d_main.c

```diff
@@ -2141,6 +2199,28 @@ void D_DoomMain(void)
             D_AddFile(myargv[p]);
     }
 
+  if (!M_CheckParm("-save"))
+  {
+    int i;
+    char *wadname = wadfiles[0], *oldsavegame = basesavegame;
+
+    for (i = mainwadfile; i < numwadfiles; i++)
```

Comment:
Should we check for the first PWAD that contains a map, or only check the first one?

Reply:
I like the current approach.